src/rtp_transceiver/mod.rs

```diff
@@ -304,28 +310,61 @@ impl RTCRtpTransceiver {
         self.kind
     }
 
-    /// direction returns the RTPTransceiver's current direction
+    /// direction returns the RTPTransceiver's desired direction.
     pub fn direction(&self) -> RTCRtpTransceiverDirection {
         self.direction.load(Ordering::SeqCst).into()
     }
 
     pub(crate) fn set_direction(&self, d: RTCRtpTransceiverDirection) {
+        let previous: RTCRtpTransceiverDirection = self.direction.load(Ordering::SeqCst).into();
```

Comment:
note: you can use https://rust.group/core/sync/atomic/struct.AtomicU8.html#method.swap to get previous value and set new one in one hop

Reply:
Good catch, this was supposed to be throw away code, but I think having the logging might be useful so I'll use `swap` instead.